src/server/GameCards.ts

```diff
@@ -92,18 +94,33 @@ export class GameCards {
     return this.getCards<StandardProjectCard>('standardProjects');
   }
   public getCorporationCards(): Array<ICorporationCard> {
-    return this.getCards<ICorporationCard>('corporationCards')
+    const cards = this.getCards<ICorporationCard>('corporationCards')
       .filter((card) => card.name !== CardName.BEGINNER_CORPORATION);
+    return this.addCustomCards(cards, this.gameOptions.customCorporationsList);
   }
   public getPreludeCards() {
-    const preludes = this.getCards<PreludeCard>('preludeCards');
+    let preludes = this.getCards<PreludeCard>('preludeCards');
     // https://github.com/terraforming-mars/terraforming-mars/issues/2833
     // Make Valley Trust playable even when Preludes is out of the game
     // by preparing a deck of preludes.
-    if (preludes.length > 0) {
-      return preludes;
+    if (preludes.length === 0) {
+      preludes = this.instantiate(PRELUDE_CARD_MANIFEST.preludeCards);
     }
-    return this.instantiate(PRELUDE_CARD_MANIFEST.preludeCards);
+    return this.addCustomCards(preludes, this.gameOptions.customPreludes);
+  }
+
+  private addCustomCards<T extends ICard>(cards: Array<T>, customList: Array<CardName> | undefined): Array<T> {
```

Comment:
This avoids having to check if `customList` is defined and creating `list` var. Typescript can generate this code.

```suggestion
  private addCustomCards<T extends ICard>(cards: Array<T>, customList: Array<CardName> = []): Array<T> {
```

Reply:
This is nice.